csrc/device_lower/pass/scalar_hoist.cpp

```diff
@@ -213,6 +221,12 @@ std::pair<Val*, bool> CommonScalarMap::hoistScalarImpl(
     create_fn(value->container(), inputs, {value}, def->attributes());
   }
 
+  // TODO: this is only a temoprary hack, will be removed in a follow-up PR
+  if (shouldHoistToHost(value)) {
+    GpuLower::current()->allKnownVals().emplace_back(value);
```

Comment:
Is this to avoid trying to hoist TensorMap vals?

Reply:
This is to request to evaluate `TensorMap` on the host, instead of generating it on device

Reply:
Thanks. This is noted as temporary, but please leave your answer as a comment if it's going to be relevant after the future cleanup.

Reply:
added